### PR TITLE
Add support for exp function and update pow function

### LINKED ISSUE #583: Document exp function and the updates to the pow function
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/7971

### Which part(s) of the docs might be affected or should be updated? And how?

SQL -> Functions and operators -> Math. Note that the decimal version of the `exp` function is not supported. 

### Reference

https://www.postgresql.org/docs/15/functions-math.html

### PR DESCRIPTION
**Pull request by bernscode**
Add support for exp function and update pow function.

## Info
- **Description**: 
Add support for exp function and update pow function.

- **Preview**: 
https://pr-598.d2fbku9n2b6wde.amplifyapp.com/docs/upcoming/sql-function-mathematical/

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/7971

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/583

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>Scroll down and open this link: image
</details>

### CODE REVIEW

File: docs/sql/functions-operators/sql-function-mathematical.md [hunk 1]
```diff
@@ -24,8 +24,9 @@ title: Mathematical functions and operators
 | ----------- | ----------- | ----------- | 
 | abs ( *input_value* ) → *absolute_value* | Returns the absolute value of *input_value*. The *input_value* can be type int or decimal. The return type is the same as the *input_value* type. | abs(-3) → 3 |
 | ceil ( *numeric_input* ) → *integer_output* <br /> ceil ( *double_precision_input* ) → *integer_output* | Returns the nearest integer greater than or equal to the argument. | ceil(1.23559) → 2 <br /> ceil(-1.23559) → -1 |
+| exp ( *numeric* ) → *output_value* | Returns the exponential value of *numeric*. | exp(2.0) → 7.38905609893065 |
```
Comment (lmatz) [inline docs/sql/functions-operators/sql-function-mathematical.md]: Probably the doc I wrote in PR made it confusing 😅 

For `exp`, we support `double precision` only, the same as `pow`.
`numeric` and `decimal` are the same data types. Currently, we cannot do `exp( numeric/decimal )`. The `numeric/decimal` in `exp` will be automatically converted to `double precision` first, and then do the computation, and the return result is also of `double precision` data type.

`numeric/decimal` has a much large data range than `double precision`.

Comment (bernscode) [reply docs/sql/functions-operators/sql-function-mathematical.md]: @lmatz That makes perfect sense now. I've made the update to the latest commit! Thanks for the clarification!

**Review by lmatz — APPROVED** [review]
LGTM

**Review by hengm3467 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**